### PR TITLE
Improve RBAC documentation

### PR DESCRIPTION
**Pull request by liggitt**
* Document new command line utilities
* Document immutable roleRef reasons
* Document cross-namespace service account permission grants

Preview at https://deploy-preview-11254--kubernetes-io-master-staging.netlify.com/docs/reference/access-authn-authz/rbac/

/sig auth
/assign @enj

### CODE REVIEW

**Comment by mistyhacks** [comment]
This looks good to me but it needs a techical reviewer. Please nominate one, @liggitt . :)

**Comment by liggitt** [comment]
@enj is the man

**Comment by liggitt** [comment]
/assign dims
can you take a look?

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 1]
```diff
@@ -735,46 +753,150 @@ To bootstrap initial roles and role bindings:
...
+* Create a `Role` named "foo" with apiGroups specified:
+
+    ```
+    kubectl create role foo --verb=get,list,watch --resource=replicasets.apps
+    ```
+
+* Create a `Role` named "foo" with subresource permissions:
+
+    ```
+    kubectl create role foo --verb=get,list,watch --resource=pods,pods/status
```
Comment (dims) [inline content/en/docs/reference/access-authn-authz/rbac.md]: Want to add an example that uses `--resource-name`?

**Comment by dims** [comment]
LGTM 👍 one question in-line

File: content/en/docs/reference/access-authn-authz/rbac.md (hunk 1 above)
Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/rbac.md]: added

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 2]
```diff
@@ -148,6 +148,24 @@ roleRef:
   apiGroup: rbac.authorization.k8s.io
 ```
 
+The `Role` or `ClusterRole` a binding object refers to may not be modified.
```
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/rbac.md]:
```suggestion
You cannot modify the `Role` or `ClusterRole` that a binding object refers to.
```
How about this wording?

Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/rbac.md]: went with "You cannot modify which `Role` or `ClusterRole` a binding object refers to." instead

**Comment by liggitt** [comment]
cc @kubernetes/sig-docs-en-reviews 

technical content is reviewed, pending docs approval

**Comment by Rajakavitha1** [comment]
reviewed the updated content at: https://deploy-preview-11254--kubernetes-io-master-staging.netlify.com/docs/reference/access-authn-authz/rbac/
/lgtm

**Comment by Rajakavitha1** [comment]
@mistyhacks could you please take a look and state your approval

**Comment by zparnold** [comment]
@liggitt Awesome job as always! 
/approve
